BSP/Simulator/DataGeneration/simulate.py

```diff
@@ -213,10 +215,46 @@ def main():
                     print("Enter the frequency you would like to change the clock to in Hz.")
                     frequency = int(input())
                     PLL.Change_Frequency(frequency)
+                elif choice == 'EEPROM':
+                    errormsg = 0
+                    print("Enter 'all' for all data or 'read' to enter specific address to read.")
+                    print(">>", end="")
+                    choiceEEPROM1 = input()
+                    print("Enter 'raw' to read the raw hex values or 'msg' for the translated error messages.", end="\n")
+                    print("If invalid response is given, default is raw data.")
+                    choiceEEPROM2 = input()
+                    if choiceEEPROM2 == 'raw':
+                        errormsg = 0
+                    elif choiceEEPROM2 == 'msg':
+                        errormsg = 1
+                    else:
+                        print("Invalid entry...", end="\n")
+                        print("Defaulted to raw data.")
+                    if choiceEEPROM1 == 'all':
+                        print(I2C.EEPROM_Dump(errormsg))
+                        print("Enter to continue simulator:")
+                        print(">>", end="")
+                        choice = input()
+                    elif choiceEEPROM1 == 'read':
+                        print("Enter address to read faults from (in hex format).")
+                        choiceEEPROM3 = input()
+                        EEPROMAddress = int(choiceEEPROM3, 16)
+                        if EEPROMAddress > 0 and EEPROMAddress < maxEEPROMAddress:
+                            print(I2C.I2C_Read(EEPROMAddress, errormsg))
+                            print("Enter to continue simulator:")
+                            choiceEEPROM = input()
+                        else:
+                            print("Invalid address...", end="\n")
+                            print("Enter to continue simulator:")
+                            choiceEEPROM = input()
+                    else:
```

Comment:
The way this is set up leads to some counterintuitive behavior. For example, if the user gives an invalid input to the first question, they will only get an "invalid entry" error message after they answer the second question. This is confusing because it seems like the "invalid entry" error message is referring to the most recent question. I think it would be better if either
1. The error message was displayed immediately after the invalid entry, or
2. The error message was more specific, so it is obvious that it refers to the first question.

Reply:
Got it, went with #2